src/databricks/labs/ucx/source_code/files.py

```diff
@@ -44,7 +50,114 @@ def __repr__(self):
         return f"<LocalFile {self._path}>"
 
 
-class LocalFileMigrator:
+class LocalDirectory(SourceContainer):
+    def __init__(self, path: Path, file_loader: FileLoader, dir_loader: DirectoryLoader):
+        self._path = path
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+
+    @property
+    def path(self) -> Path:
+        return self._path
+
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
+        return list(self._build_dependency_graph(parent))
+
+    def _build_dependency_graph(self, parent: DependencyGraph) -> Iterable[DependencyProblem]:
+        for child_path in self._path.iterdir():
+            if child_path.is_dir():
+                dependency = Dependency(self._dir_loader, child_path)
+                yield from parent.register_dependency(dependency).problems
+                continue
+            dependency = Dependency(self._file_loader, child_path)
+            yield from parent.register_dependency(dependency).problems
+
+    def __repr__(self):
+        return f"<LocalDirectory {self._path}>"
+
+
+# TODO remove duplicate code by inheriting from LocatedAdvice once https://github.com/databrickslabs/lsql/issues/103 is fixed
+# pylint: disable=duplicate-code
+@dataclass
+class FileProblem:
+    path: str
+    code: str
+    message: str
+    start_line: int
+    start_col: int
+    end_line: int
+    end_col: int
+
+
+class LocalFilesLinter:
+
+    def __init__(
+        self,
+        languages: Languages,
+        file_loader: FileLoader,
+        dir_loader: DirectoryLoader,
+        path_lookup: PathLookup,
+        dependency_resolver: DependencyResolver,
+    ) -> None:
+        self._languages = languages
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+        self._path_lookup = path_lookup
+        self._dependency_resolver = dependency_resolver
+        self._extensions = {".py": Language.PYTHON, ".sql": Language.SQL}
+
+    def lint(self, path: Path) -> list[FileProblem]:
+        return list(self._lint(path))
+
+    def _lint(self, path: Path) -> Iterable[FileProblem]:
+        loader = self._dir_loader if path.is_dir() else self._file_loader
+        dependency = Dependency(loader, path)
+        graph = DependencyGraph(dependency, None, self._dependency_resolver, self._path_lookup)
+        container = dependency.load(self._path_lookup)
+        assert container is not None  # because we just created it
+        dependency_problems = container.build_dependency_graph(graph)
+        if dependency_problems:
+            problems = [
```

Comment:
list comprehension doesn't make it more readable here, use simple for loop

Reply:
done